utils/wasm-gen/src/tests.rs

```diff
@@ -199,8 +226,69 @@ fn error_processing_works_for_fallible_syscalls() {
         );
 
         // Assert that syscall results will be ignored.
-        let termination_reason =
-            execute_wasm_with_syscall_injected(syscall, true, params_config, initial_memory_write);
+        let termination_reason = execute_wasm_with_custom_configs(
+            &mut unstructured2,
+            sys_calls_config_builder.build(),
+            initial_memory_write.clone(),
+            INITIAL_PAGES,
+            context_settings,
+            true,
+        );
+
+        assert_eq!(
+            termination_reason,
+            TerminationReason::Actor(ActorTerminationReason::Success),
+            "syscall: {}",
+            syscall.to_str()
+        );
+    }
+}
+
+#[test]
+fn precise_syscalls_works() {
```

Comment:
I think this logics can be shared with the one used in fn above, counting what I've written in the other review comments